[HPR-574] Add documentation for configuring Packer for HCP Packer (#12022)
hashicorp/packer: website/content/docs/configure.mdx

  and the [`packer init`](/docs/commands/init) command to install plugins; if
  you are using both, the `required_plugin` config will take precedence.

### HCP Packer Configuration

You can configure both legacy JSON and HCL2 Packer templates to publish image metadata to an active HCP Packer registry. The HCP Packer registry helps you track information about machine images, clearly designate which images are appropriate for test and production environments, and query the right images to use in both Packer and Terraform configurations.

> For complete configuration details and examples, refer to [Packer Template Configuration](https://cloud.hashicorp.com/docs/packer/store-image-metadata/template-configuration) in the HCP Packer documentation.

Author: Once the updated template configuration page has been merged this will need to be updated.

> For complete configuration details and examples, refer to [Packer Template Configuration](https://cloud.hashicorp.com/docs/packer/store-image-metadata/template-configuration) in the HCP Packer documentation.

Author: ```suggestion
For complete configuration details and examples, refer to [Packer Template Configuration](https://cloud.hashicorp.com/docs/packer/store-image-metadata/packer-template-configuration) in the HCP Packer documentation.
```

  and the [`packer init`](/docs/commands/init) command to install plugins; if
  you are using both, the `required_plugin` config will take precedence.

> ### Packer configuration for HCP Packer

Reviewer: Should this be an H3? It almost feels like it should be an H2 with the others

```suggestion
### HCP Packer Configuration
```

>   Packer can be configured to publish artifact metadata for all completed builds to an active HCP Packer registry.
>   The configuration for HCP Packer is managed by environment variables for legacy JSON templates or via a `hcp_packer_registry`
>   block for HCL2 templates. 
> 
>   For complete details on how to configure Packer for the HCP Packer registry see [Configuring Packer for HCP Packer ](docs/configure-hcp-packer)

Reviewer: My opinion is that we link to the HCP Packer docs for this. We do such a good job there of like....giving you all the info and examples. Thoughts?

```suggestion
You can configure both legacy JSON and HCL2 Packer templates to publish image metadata to an active HCP Packer registry. The HCP Packer registry helps you track information about machine images, clearly designate which images are appropriate for test and production environments, and query the right images to use in both Packer and Terraform configurations.

For complete configuration details and examples, refer to [Packer Template Configuration](https://cloud.hashicorp.com/docs/packer/store-image-metadata/template-configuration) in the HCP Packer documentation.
```

Author: I like this idea if we make the new Configuring HCP Packer page the HCP Packer page. The sidebar in the image below points users to this new page. Otherwise it seems confusing to have a "Configuring HCP Packer" link but in the Configuring Packer page we point to a different set of docs. What do you think? 

<img width="248" alt="Screen Shot 2022-10-20 at 12 19 08 PM" src="https://user-images.githubusercontent.com/1749304/197003779-848c6b7d-ff07-4583-b0a0-2ba7364208a8.png">
